File: Backspace String Compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        count=0
        s=list(s)
        t=list(t)
        for i in range(len(s)-1,-1,-1):
            if s[i]=="#":
                count+=1
                s[i]=''
            else:
                if count!=0:
                    s[i]=''
                    count-=1
                else:
                    pass
        count=0
        for j in range(len(t)-1,-1,-1):
            if t[j]=="#":
                count+=1
                t[j]=''
            else:
                if count!=0:
                    t[j]=''
                    count-=1
                else:
                    pass

        return "".join(s)=="".join(t)

File: Java/Backspace String Compare.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        StringBuilder sb1 = new StringBuilder();
        StringBuilder sb2 = new StringBuilder();
        int counter=0;
        for(int i = s.length()-1;i>=0;i--){
            if(s.charAt(i)=='#'){
                counter++;
            }
            else{
                if(counter>0){
                    counter--;
                }
                else{
                    sb1.append(s.charAt(i));
                }
            }

        }
        counter=0;
        for(int i = t.length()-1;i>=0;i--){
            if(t.charAt(i)=='#'){
                counter++;
            }
            else{
                if(counter>0){
                    counter--;
                }
                else{
                    sb2.append(t.charAt(i));
                }
            }

        }
        return sb1.toString().equals(sb2.toString());


        
    }
}
